test/e2e/workable.go

```diff
@@ -0,0 +1,319 @@
+	return nil
+}
+
+func testKubeSchedulerPort() error {
+	proto := "https"
+	client := getHttpClient()
+	resp, err := client.Get(fmt.Sprintf("%s://127.0.0.1:10250/healthz", proto))
+	if err != nil {
+		return fmt.Errorf("kube scheduler connection error: %w", err)
+	}
+	defer func() {
+		if closeErr := resp.Body.Close(); closeErr != nil && err == nil {
+			err = closeErr
+		}
+	}()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	if string(body) != "ok" {
+		fmt.Println("Error: kube scheduler connection")
+		fmt.Println("curl -s " + proto + "://127.0.0.1:10250/healthz")
+		fmt.Println(string(body))
+		return fmt.Errorf("kube scheduler connection error")
+	}
+
+	return nil
+}
+
+func testKubeControllerManagerPort() error {
+	proto := "https"
+	client := getHttpClient()
+	resp, err := client.Get(fmt.Sprintf("%s://127.0.0.1:10260/healthz", proto))
+	if err != nil {
+		return fmt.Errorf("kube controller manager connection error: %w", err)
+	}
+	defer func() {
+		if closeErr := resp.Body.Close(); closeErr != nil && err == nil {
+			err = closeErr
+		}
+	}()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	if string(body) != "ok" {
+		fmt.Println("Error: kube controller manager connection")
+		fmt.Println("curl -s " + proto + "://127.0.0.1:10260/healthz")
+		fmt.Println(string(body))
+		return fmt.Errorf("kube controller manager connection error")
+	}
+
+	return nil
+}
+
+func getHttpClient() *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, // #nosec G402
+		},
+	}
+}
+
+func retry(f func() error) error {
+	for i := 0; i < 120; i++ {
+		if err := f(); err == nil {
+			return nil
+		}
+		time.Sleep(1 * time.Second)
+	}
+	return fmt.Errorf("retry failed after 120 attempts")
+}
+
+func CaseWorkable(kwokctlPath, clusterName, clusterRuntime string) *features.FeatureBuilder {
+	return features.New("Workable").
+		Assess("Test Workable", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			ctxName := "kwok-" + clusterName
+			err := exec.Command("kubectl", "config", "use-context", ctxName).Run() // #nosec G204
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = testWorkable(kwokctlPath, clusterName)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if clusterRuntime != "kind" && clusterRuntime != "kind-podman" {
+				if err = testKubeControllerManagerPort(); err != nil {
+					t.Fatal(err)
+				}
+				if err = testKubeSchedulerPort(); err != nil {
+					t.Fatal(err)
+				}
+				if err = testEtcdPort(); err != nil {
+					t.Fatal(err)
+				}
+			}
+			if runtime.GOOS != "windows" {
+				if err = testEtcdctlGet(kwokctlPath, clusterName); err != nil {
+					t.Fatal(err)
+				}
+			}
+			if err = testPrometheus(); err != nil {
+				t.Fatal(err)
+			}
+			if err = testJaeger(); err != nil {
+				t.Fatal(err)
+			}
+			if err = testKwokControllerPort(); err != nil {
+				t.Fatal(err)
+			}
+			return ctx
```

Comment:
We should make sure that the test style is the same
- each step is wrapped in Assess()
- using the same retry mechanism as the other tests
- Use (*testing.T).Log instead fmt.Println

Reply:
@wzshiming  By each step do you mean each function i.e testWorkable,testKubeSchedulerPort,testPrometheus,testJaeger,etc. should get a separate assess step?